module/VuFind/src/VuFind/ILS/Driver/AbstractBase.php

```diff
@@ -66,6 +70,170 @@ public function setConfig($config)
+     * @param string  $cat_password     The password used to log in
+     * @param ?string $email            The patron's email address (null if unavailable)
+     * @param string  $firstname        The patron's first name
+     * @param string  $lastname         The patron's last name
+     * @param ?string $major            The patron's major (null if unavailable)
+     * @param ?string $college          The patron's college (null if unavailable)
+     * @param array   $nonDefaultFields Non default fields not documented in the documentation.
+     *                                  Merges into the resulting patron array.
+     *
+     * @see https://vufind.org/wiki/development:plugins:ils_drivers#patronlogin
+     *
+     * @return array
+     */
+    public function createPatronArray(
+        string $id,
+        string $cat_username = '',
+        string $cat_password = '',
+        ?string $email = null,
+        string $firstname = '',
+        string $lastname = '',
+        ?string $major = null,
+        ?string $college = null,
+        array $nonDefaultFields = []
+    ): array {
+        $patron = compact(
+            'id',
+            'email',
+            'firstname',
+            'lastname',
+            'major',
+            'college'
+        );
+        $this->debugDriverResult(__FUNCTION__, $patron, $nonDefaultFields);
+        // Merge non default fields into the resulting patron array
+        if ($nonDefaultFields) {
+            $patron = array_merge($patron, $nonDefaultFields);
+        }
+        // Add cat_username and cat_password after debugging to avoid logging these values into a log file
+        $patron['cat_username'] = $cat_username;
+        $patron['cat_password'] = $cat_password;
+        return array_map(fn ($val) => is_string($val) ? trim($val) : $val, $patron);
+    }
+
+    /**
+     * Create a profile array according to getMyProfile specs defined in the documentation.
+     * Each value is trimmed if they are not null.
+     *
+     * @param ?string                        $firstname        Profile first name
+     * @param ?string                        $lastname         Profile last name
+     * @param string                         $birthdate        Y-m-d or an empty string
+     * @param ?string                        $address1         Address 1
+     * @param ?string                        $address2         Address 2
+     * @param ?string                        $city             City
+     * @param ?string                        $country          Country
+     * @param ?string                        $zip              Postal code
+     * @param ?string                        $phone            Phone number
+     * @param ?string                        $mobile_phone     Mobile phone number
+     * @param ?string                        $expiration_date  Profile expiration date
+     * @param TranslatableString|string|null $group            Group i.e. Student, Staff, Faculty, etc
```

Comment:
Maybe it would be more flexible to use `Stringable` instead of `TranslatableString`. And maybe it's worth considering whether we should standardize on `Stringable|string|null` across the board for future flexibility...

```suggestion
     * @param Stringable|string|null $group            Group i.e. Student, Staff, Faculty, etc
```